folly/Replaceable.h

```diff
@@ -640,8 +640,7 @@ class alignas(T) Replaceable
   aligned_storage_for_t<T> storage_[1];
 };
 
-#if __cplusplus > 201402L
-// C++17 allows us to define a deduction guide:
+#if __cpp_deduction_guides >= 201703
```

Comment:
Seems like MSVC2017 does not define this name but does support deduction guides. Can guard this with `|| _MSC_VER`. (We don't support earlier versions of MSVC anymore.)

Reply:
That's a bit annoying from a QoI perspective. Based on the `pushmi` usages of `__cpp_deduction_guides`, we may not support things in `experimental` with MSVC?

I can use `|| _MSC_VER` here, but we may want to put this expression into a `constexpr bool` in `Portability.h` after we acquire some more uses of deduction guides.